releases/release-4.0.16.md

```diff
@@ -0,0 +1,108 @@
+---
+title: TiDB 4.0.16 Release Notes
+---
+
+# TiDB 4.0.16 Release Notes
+
+Release Date: December 10, 2021
+
+TiDB version: 4.0.16
+
+## Compatibility changes
+
++ TiKV
+
+    - Before v4.0.16, when TiDB converts an illegal UTF-8 string to a Real type, an error is reported directly. Starting from v4.0.16, TiDB processes the conversion according to the legal UTF-8 prefix in the string [#11466](https://github.com/tikv/tikv/issues/11466)
+
++ Tools
+
+    + TiCDC
+
+        - Change the default value of Kafka Sink `max-message-bytes` to 1 MB to prevent TiCDC from sending too large messages to Kafka clusters [#2962](https://github.com/pingcap/ticdc/issues/2962)
+        - Change the default value of Kafka Sink `partition-num` to 3 so that TiCDC distributes messages across Kafka partitions more evenly [#3337](https://github.com/pingcap/ticdc/issues/3337)
+
+## Improvements
+
++ TiKV
+
+    - Reduce the disk space usage by using the zstd algorithm to compress SST files when using Backup & Restore to restore data or using TiDB Lightning's Local-backend [#11469](https://github.com/tikv/tikv/issues/11469)
+
++ Tools
+
+    + Backup & Restore (BR)
+
+        - Improve the robustness for restoring [#27421](https://github.com/pingcap/tidb/issues/27421)
+
+    + TiCDC
+
+        - Add a Tick frequency limit to EtcdWorker to prevent frequent Etcd writes from affecting PD services [#3112](https://github.com/pingcap/ticdc/issues/3112)
```

Comment:
```suggestion
        - Add a tick frequency limit to EtcdWorker to prevent frequent etcd writes from affecting PD services [#3112](https://github.com/pingcap/ticdc/issues/3112)
```